src/enzyme_ad/jax/Utils.h

```diff
@@ -354,7 +356,7 @@ template <typename Child> class GuaranteedResultAnalysisBase {
 
     // Map of operations we have seen before. The target of the map[o] is a list
     // of sub-queries, that if all true prove that `o` is no-nan.
-    DenseMap<Operation *, SmallPtrSet<Operation *, 2>> seen;
+    llvm::MapVector<Operation *, llvm::SmallSetVector<Operation *, 2>> seen;
```

Comment:
looking at the code, this doesnt need to be a mapvector, since seen is not used non determinstically